syntax = "proto3";

message RPC {
    string function_name = 1;
    Args args = 2;
}

message channel_data {
   repeated int32 values= 1; 
}

message channel_data2 {
    repeated float values = 1; 
}

message prepost {
    repeated int32 value = 1;
}

message reply {
    string value = 1;
}

message Args {
    string unique_ADC_name = 11;
    string addr = 10;
    int32 port = 9;
    int32 number_of_channels = 8;
    repeated int64 timestamps = 7;
    int32 presamples = 6;
    int32 postsamples = 5;
    map<string, channel_data> data = 4;

    map<int32, channel_data2> data2 = 3;
    map<int32, prepost> pre_post = 2;
    map<int32, string> offsets = 1;
}


service RPC_req {
    rpc register_ADC(Args) returns (reply) {}
}
